shared/extension/Configuration.hpp

```diff
@@ -217,42 +223,45 @@ namespace extension {
 // NUClear configuration extension
 namespace NUClear::dsl {
     namespace operation {
+        namespace {
+            [[nodiscard]] std::string getFirstCommandLineArg() {
+                std::shared_ptr<const message::CommandLineArguments> args =
+                    store::DataStore<message::CommandLineArguments>::get();
+                std::vector<char> data(args->at(0).cbegin(), args->at(0).cend());
+                data.push_back('\0');
+                return std::string(data.data());
```

Comment:
Also, don't assume that `CommandLineArguments` has something in it, it is possible for it to be empty
```suggestion
                return std::string(args->at(0).cbegin(), args->at(0).cend());
```

Reply:
Pretty sure it's not possible for it to be empty? You always at least have the actual binary name itself that was executed (which is what this is accessing).

I suppose if you made a custom PowerPlant instance rather than via the roles system you could do it?

Reply:
When running unit tests the powerplant is started in an interesting way

Reply:
Thanks for pointing this out @Bidski, that's exactly what I came to configuration to do (and promptly forgot 😂), precisely for that reason

Reply:
This is still ugly and can be made not ugly

Reply:
It has been beautified :heart_eyes: